src/analysis/inner_product_space/adjoint.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2021 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis, Heather Macbeth
+-/
+
+import analysis.inner_product_space.dual
+
+/-!
+# Adjoint of operators on Hilbert spaces
+
+Given an operator `A : E →L[𝕜] F`, where `E` and `F` are Hilbert spaces, its adjoint
+`adjoint A : F →L[𝕜] E` is the unique operator such that `⟪x, A y⟫ = ⟪adjoint A x, y⟫` for all
+`x` and `y`.
+
+We then use this to put a star algebra structure on `E →L[𝕜] E` with the adjoint as the star
+operation.
+
+## Implementation notes
+
+* The adjoint is defined as a conjugate-linear isometric equivalence between `E →L[𝕜] F` and
+  `F →L[𝕜] E`. The bare function `adjoint'` is only an intermediate definition and is not meant
+  to be used outside this file.
```

Comment:
I would be inclined to call it `adjoint_aux` to make this obvious. Usually we use the primed names for variations of an API that users might actually want.

Reply:
Fair enough.